libraries/AP_Compass/AP_Compass.h

```diff
@@ -117,8 +117,20 @@ friend class AP_Compass_Backend;
     void save_offsets(uint8_t i);
     void save_offsets(void);
 
-    // return the number of compass instances
-    uint8_t get_count(void) const { return _compass_count; }
+    /// Check if the compass instance was registered and available for use
+    ///
+    /// @param instance			compass instance
+    ///
+    /// @returns true if compass backend successfully registered
+    ///
+    bool instance_available(uint8_t i) const { return _state[i].registered; }
```

Comment:
given how this is used I'd like it to be a cpp function and it also check if i >= COMPASS_MAX_INSTANCES, to prevent index beyond end of array when we limit the number of compasses below 3 on some builds